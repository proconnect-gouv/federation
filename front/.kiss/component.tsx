import React from 'react';

interface DumbComponentProps {
  className: string;
}

export const DumbComponent: React.FC<DumbComponentProps> = React.memo(
  ({ className }: DumbComponentProps) => {
    return <div />;
  },
);

DumbComponent.displayName = 'DumbComponent';
