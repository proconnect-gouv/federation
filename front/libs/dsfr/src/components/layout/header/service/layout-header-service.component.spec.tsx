// @see _doc/jest.md
import { render } from '@testing-library/react';

import type { HeaderService } from '../../../../interfaces';
import { LayoutHeaderServiceComponent } from './layout-header-service.component';

describe('LayoutHeaderServiceComponent', () => {
  const headerServiceMock: HeaderService = {
    name: 'any service name mock',
    title: 'title service mock',
  };

  it('should match the snapshot', () => {
    // when
    const { container } = render(<LayoutHeaderServiceComponent service={headerServiceMock} />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot when baseline is defined', () => {
    // given
    const headerServiceMockWithBaseline = {
      ...headerServiceMock,
      baseline: 'any baseline mock',
    };

    // when
    const { container } = render(
      <LayoutHeaderServiceComponent service={headerServiceMockWithBaseline} />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot when href is defined', () => {
    // given
    const headerServiceMockWithHref = {
      ...headerServiceMock,
      href: '/any-href-mock',
    };

    // when
    const { container } = render(
      <LayoutHeaderServiceComponent service={headerServiceMockWithHref} />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot when baseline and href are defined', () => {
    // given
    const headerServiceMockWithBaselineAndHref = {
      ...headerServiceMock,
      baseline: 'any baseline mock',
      href: '/any-href-mock',
    };

    // when
    const { container } = render(
      <LayoutHeaderServiceComponent service={headerServiceMockWithBaselineAndHref} />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should service name be in the document with a title', () => {
    // when
    const { getByText, getByTitle } = render(
      <LayoutHeaderServiceComponent service={headerServiceMock} />,
    );
    const elementName = getByText('any service name mock');
    const elementTitle = getByTitle('title service mock');

    // then
    expect(elementName).toBeInTheDocument();
    expect(elementTitle).toBeInTheDocument();
  });

  it('should service name be in the document with a title and when href is not defined', () => {
    // when
    const { getByText, getByTitle } = render(
      <LayoutHeaderServiceComponent service={headerServiceMock} />,
    );
    const elementName = getByText('any service name mock');
    const elementTitle = getByTitle('title service mock');

    // then
    expect(elementName).toBeInTheDocument();
    expect(elementTitle).toBeInTheDocument();
    expect(elementTitle.getAttribute('href')).toBe('/');
  });

  it('should service name be in the document with a title and when href is defined', () => {
    // given
    const headerServiceMockWithHref = {
      ...headerServiceMock,
      href: '/any-href-mock',
    };

    // when
    const { getByText, getByTitle } = render(
      <LayoutHeaderServiceComponent service={headerServiceMockWithHref} />,
    );
    const elementName = getByText('any service name mock');
    const elementTitle = getByTitle('title service mock');

    // then
    expect(elementName).toBeInTheDocument();
    expect(elementTitle).toBeInTheDocument();
    expect(elementTitle.getAttribute('href')).toBe('/any-href-mock');
  });

  it('should service baseline be in the document', () => {
    // given
    const headerServiceMockWithBaseline = {
      ...headerServiceMock,
      baseline: 'acme service baseline mock',
    };

    // when
    const { getByText } = render(
      <LayoutHeaderServiceComponent service={headerServiceMockWithBaseline} />,
    );
    const element = getByText('acme service baseline mock');

    // then
    expect(element).toBeInTheDocument();
  });
});
