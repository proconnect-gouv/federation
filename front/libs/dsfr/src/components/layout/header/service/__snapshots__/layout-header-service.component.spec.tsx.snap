// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutHeaderServiceComponent should match the snapshot 1`] = `
<div>
  <div
    class="fr-header__service"
    data-testid="layout-header-service-component"
  >
    <a
      href="/"
      title="title service mock"
    >
      <p
        class="fr-header__service-title"
        data-testid="layout-header-service-component-name"
      >
        any service name mock
      </p>
    </a>
  </div>
</div>
`;

exports[`LayoutHeaderServiceComponent should match the snapshot when baseline and href are defined 1`] = `
<div>
  <div
    class="fr-header__service"
    data-testid="layout-header-service-component"
  >
    <a
      href="/any-href-mock"
      title="title service mock"
    >
      <p
        class="fr-header__service-title"
        data-testid="layout-header-service-component-name"
      >
        any service name mock
      </p>
    </a>
    <p
      class="fr-header__service-tagline"
      data-testid="layout-header-service-component-baseline"
    >
      any baseline mock
    </p>
  </div>
</div>
`;

exports[`LayoutHeaderServiceComponent should match the snapshot when baseline is defined 1`] = `
<div>
  <div
    class="fr-header__service"
    data-testid="layout-header-service-component"
  >
    <a
      href="/"
      title="title service mock"
    >
      <p
        class="fr-header__service-title"
        data-testid="layout-header-service-component-name"
      >
        any service name mock
      </p>
    </a>
    <p
      class="fr-header__service-tagline"
      data-testid="layout-header-service-component-baseline"
    >
      any baseline mock
    </p>
  </div>
</div>
`;

exports[`LayoutHeaderServiceComponent should match the snapshot when href is defined 1`] = `
<div>
  <div
    class="fr-header__service"
    data-testid="layout-header-service-component"
  >
    <a
      href="/any-href-mock"
      title="title service mock"
    >
      <p
        class="fr-header__service-title"
        data-testid="layout-header-service-component-name"
      >
        any service name mock
      </p>
    </a>
  </div>
</div>
`;
